### ISSUE TITLE
Importing is very slow

### ISSUE THREAD

**Issue by carlosperate** [MarshalX/atproto#453]
In my dev environment  (Macbook Pro 2020 i7) importing `atproto` takes over 3 seconds with Python 3.10, 3.11 and 3.12.

I couldn't find mentions of this in the issue tracker, so I assume this is not as expected?

```
$ python --version
Python 3.12.1

$ time python -c "from atproto import Client"
________________________________________________________
Executed in    3.63 secs    fish           external
   usr time    3.00 secs    0.14 millis    3.00 secs
   sys time    0.32 secs    1.23 millis    0.32 secs
```

```
$ python --version
Python 3.11.0

$ time python -c "from atproto import Client"
________________________________________________________
Executed in    4.20 secs    fish           external
   usr time    3.50 secs  136.00 micros    3.50 secs
   sys time    0.34 secs  954.00 micros    0.34 secs
```

```
$ python --version
Python 3.10.8

$ time python -c "from atproto import Client"
________________________________________________________
Executed in    3.97 secs    fish           external
   usr time    3.60 secs    0.12 millis    3.60 secs
   sys time    0.22 secs    1.52 millis    0.21 secs
```

The Python 3.12 venv pip list output:
```
Package           Version
----------------- -----------
annotated-types   0.7.0
anyio             4.6.2.post1
atproto           0.0.55
certifi           2024.8.30
cffi              1.17.1
click             8.1.7
cryptography      43.0.3
dnspython         2.7.0
h11               0.14.0
httpcore          1.0.7
httpx             0.27.2
idna              3.10
libipld           3.0.0
pip               23.2.1
pycparser         2.22
pydantic          2.10.1
pydantic_core     2.27.1
sniffio           1.3.1
typing_extensions 4.12.2
websockets        13.1
```

**Comment by MarshalX** [comment]
This is know one :( and comes from pydantic https://github.com/MarshalX/atproto/discussions/441

**Comment by MarshalX** [comment]
Try “from atproto import models”. Should be the same bad result. Client itself is not a problem

**Comment by carlosperate** [comment]
Ah, sorry, I didn't realised there was a Discussions section in the repo as well.
Could the model imports be lazy loaded? Or is this something we need to live with until Pydantic had a chance to improve performance?

**Comment by endlisnis** [comment]
I have also noticed this.  On my very fast dev system it takes 1.3 seconds, and on my raspberry pi 5, it takes about 4 seconds.

That's crazy slow.

**Comment by MarshalX** [comment]
pydantic getting serious about import time now! they reported significant improvements in the new alpha version. feel free to test. more info: https://github.com/pydantic/pydantic/issues/9908#issuecomment-2625417484

**Comment by MarshalX** [comment]
pydantic team did a very good job in the v2.11 release. the importing time and memory usage has been improved! 

- their release with details: https://pydantic.dev/articles/pydantic-v2-11-release
- my comparison with models of atproto and bluesky: https://github.com/MarshalX/pydantic-memory-usage/commit/f3f2d3600ba1148d4abbf8961a128f028c8fdb9a

i highly recommend updating pydantic; gonna close the issue

**Comment by p1timmy** [comment]
Looks like the 0.0.68 update made startup times 3x slower, time to reopen the issue I guess?

atproto 0.0.65:

```zsh
❯ time python -c 'import atproto'
python -c 'import atproto'  1.85s user 0.24s system 97% cpu 2.143 total
```

atproto 0.0.68:

```zsh
❯ time python -c 'import atproto'
python -c 'import atproto'  3.57s user 0.40s system 98% cpu 4.010 total
```

Testing done with Python 3.14 and Pydantic 2.13.4 on a Linux desktop with Ryzen 5 5600X

**Comment by MarshalX** [comment]
Probably because of months of new lexicons, which produced tons of new pydantic models. Time to think of making some lexicons optional 🧐 like tool ozon, standard site, and maybe even bsky :) The focus must be on atproto. The rest is optional and installed on demand

**Comment by p1timmy** [comment]
Yeah would love to see lazy loading required models getting implemented, it'll help a lot with most use cases where loading all of them is overkill

**Comment by MarshalX** [comment]
@p1timmy done in https://github.com/MarshalX/atproto/pull/679. Could you please test it as well? I gonna merge it but postpone it with the release for a while. Just to make sure that it is compatible 

To install: `pip install "git+https://github.com/MarshalX/atproto.git@main"`

**Comment by p1timmy** [comment]
Much better, it's now cut down to half a second on my end! 😮

```zsh
❯ time python -c 'import atproto'
python -c 'import atproto'  0.53s user 0.06s system 96% cpu 0.615 total
```

Is that all I need to test?

**Comment by MarshalX** [comment]
@p1timmy thank you! That would be cool if you could run your real project with the new SDK version. Just to check that importing of only the required portions is working. BTW, the memory consumption should be much lower as well.

**Comment by MarshalX** [comment]
The lazy loading has been released in v0.0.69
